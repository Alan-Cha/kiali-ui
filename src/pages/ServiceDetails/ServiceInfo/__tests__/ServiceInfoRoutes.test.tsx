import * as React from 'react';
import { shallow } from 'enzyme';
import { ServiceInfoRoutes } from '../index';

const dependencies: Map<string, string[]> = new Map([['v1', ['productpage.istio-system/v1']]]);

describe('#ServiceInfoRoutes render correctly with data', () => {
  it('should render service routes', () => {
    const wrapper = shallow(<ServiceInfoRoutes dependencies={dependencies} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
