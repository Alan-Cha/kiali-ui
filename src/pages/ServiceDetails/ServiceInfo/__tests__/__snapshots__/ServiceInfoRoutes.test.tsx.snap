// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ServiceInfoRoutes render correctly with data should render service routes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServiceInfoRoutes
    dependencies={
      Map {
        "v1" => Array [
          "productpage.istio-system/v1",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        bsClass="row"
        className="row-cards-pf"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
          sm={12}
          xs={12}
        />
      </Row>,
      "className": "card-pf",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
          sm={12}
          xs={12}
        />,
        "className": "row-cards-pf",
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [],
          "componentClass": "div",
          "lg": 12,
          "md": 12,
          "sm": 12,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          bsClass="row"
          className="row-cards-pf"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            lg={12}
            md={12}
            sm={12}
            xs={12}
          />
        </Row>,
        "className": "card-pf",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            lg={12}
            md={12}
            sm={12}
            xs={12}
          />,
          "className": "row-cards-pf",
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [],
            "componentClass": "div",
            "lg": 12,
            "md": 12,
            "sm": 12,
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
